kernel/enter.c

```diff
@@ -71,6 +72,67 @@ long myst_tcall(long n, long params[6])
     return ret;
 }
 
+/**
+ * Create target directory if do not exist
+ * Then call mount syscall
+ */
+static int _create_and_mount(
+    const char* source,
+    const char* target,
+    const char* fs_type,
+    unsigned long mountflags,
+    const void* data,
+    bool is_auto)
+{
+    int ret = 0;
+    struct locals
+    {
+        myst_path_t normalized_target;
+        myst_path_t suffix;
+    };
+    struct locals* locals = NULL;
+    myst_fs_t* parent_fs;
+
+    if (!(locals = malloc(sizeof(struct locals))))
+        ERAISE(-ENOMEM);
+
+    ECHECK(myst_realpath(target, &locals->normalized_target));
+
+    // target has to be an absolute path
+    if (locals->normalized_target.buf[0] != '/')
+        ERAISE(-EINVAL);
+
+    ECHECK(myst_mount_resolve(
+        locals->normalized_target.buf, locals->suffix.buf, &parent_fs));
+
+    // check if target exists and is directory
+    // if not, create target directory
+    {
+        struct stat buf;
+        int errornum =
+            (*parent_fs->fs_stat)(parent_fs, locals->suffix.buf, &buf);
+
+        if (errornum == -ENOENT)
+            // target do not exist, create directory
+            ECHECK(myst_syscall_mkdir(locals->suffix.buf, 0777));
```

Comment:
Isn't `locals->suffix` a path relative to where the filesystem to which target belongs is mounted?

Reply:
For the case where target belongs to rootfs it doesn't make a difference. But if the target(say `/mnt/target`) is on a filesystem mounted at some path(say `/mnt`). Then `myst_mount_resolve` will return just `/target` as the suffix.

Why don't you directly use `normalized_target` here?